### PR TITLE
feat: Extend fork activation blocks data with Arbitrum data

### LINKED ISSUE #7098: Extend fork activation blocks data
### Describe the feature

Extend https://github.com/paradigmxyz/reth/blob/main/crates/ethereum-forks/src/hardfork.rs with fork activation block numbers data for more chains.

This data is non-trivial to obtain, so comments/links to lists of hardfork->block mappings are appreciated as well even if you have no plans for creating a PR with those.

Especially useful would be data for Arbitrum and any other chains enabling Shanghai/Cancun support in the near future.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by altugbakan**
Fixes #7098 

This PR aims to extend fork activations block data with Arbitrum hard fork values. Since Arbitrum upgrades happen on a timestamp-based manner, the first block on the update timestamp is supplied as the block value. ArbOS 11 is the upgrade for including Shanghai, and ArbOS 20 is the upgrade for including Cancun.

I've found some irregularities between the provided timestamps and the on-chain data, but my questions to the Arbitrum team was not answered, so I went with the provided dates by the Arbitrum foundation.

I have put these values behind a feature flag called "arbitrum", please let me know if this is not needed.

The provided update timestamps can be found in https://docs.arbitrum.foundation/network-upgrades#network-activation-statuses.

### CODE REVIEW

**Comment by gakonst** [comment]
We don't support Arbitrum yet so not sure we should include this?

**Review by mattsse — COMMENTED** [review]
thanks for this!

I need to figure out how to abstract away the cfgs, this is kinda related to revm evm setup

need to think about this a bit

File: crates/ethereum-forks/src/hardfork.rs [hunk 1]
```diff
@@ -61,11 +61,17 @@ pub enum Hardfork {
...
+    #[cfg(feature = "arbitrum")]
+    ArbOS11,
     /// Cancun.
     Cancun,
     /// Ecotone
     #[cfg(feature = "optimism")]
     Ecotone,
+    /// ArbOS 20 Atlas
+    #[cfg(feature = "arbitrum")]
+    ArbOS20Atlas,
```
Comment (mattsse) [inline crates/ethereum-forks/src/hardfork.rs:74]: I'd like to not add these and instead add these as a comment that these exists on arbitrum

File: crates/ethereum-forks/src/hardfork.rs [hunk 2]
```diff
@@ -164,6 +180,62 @@ impl Hardfork {
...
+            Hardfork::MuirGlacier => Some(0),
+            Hardfork::Berlin => Some(0),
+            Hardfork::London => Some(0),
+            Hardfork::ArrowGlacier => Some(0),
+            Hardfork::GrayGlacier => Some(0),
+            Hardfork::Paris => Some(0),
+            Hardfork::Shanghai => Some(10653737),
+            Hardfork::ArbOS11 => Some(10653737),
+            Hardfork::Cancun => Some(18683405),
+            Hardfork::ArbOS20Atlas => Some(18683405),
```
Comment (mattsse) [inline crates/ethereum-forks/src/hardfork.rs:206]: we can keep arb hardforks as comment

**Review by mattsse — CHANGES_REQUESTED** [review]
okay, I thought about this a bit,

I don't want to introduce new arb hardfork variants and features the op hardforks are bad enough.

instead we should convert them into comments and only keep the regular eth hardforks in the match stmt

**Comment by altugbakan** [comment]
Hey! I've updated the PR, but is it as you requested? Should I comment out all code about Arbitrum or not?

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
thanks, I resolved the conflicts and merged in #7305 

wasn't able to push to your main directly

**Pull request closed without merging**